modules/dnn/src/dnn.cpp

```diff
@@ -1141,6 +1144,14 @@ static Ptr<BackendWrapper> wrapMat(int backendId, int targetId, cv::Mat& m)
 
 static int g_networkId = 0;
 
+G_TYPED_KERNEL(GLayer, <GMat(GMat, cv::Ptr<LayerData>)>, "org.opencv.dnn.layer")
+{
+    static GMatDesc outMeta(GMatDesc in, cv::Ptr<LayerData> )
+    {
+        return in;
```

Comment:
Not sure if this method is correct...

The goal of `outMeta` is to let G-API know how to allocate the internal buffers to hold the results.

If your `GLayer` operation is a generic one, I assume the output dimensions change for DNN operators thorough the graph. I am wondering how it works now, but it seems you need to update the return value `GMatDesc` based on dimensions of the input (`in`) and details about the layer (`LayerData`)

Reply:
This is not relying on G-API to pass data between layers, yet.  I agree that it is better implemented that way.  Please see my comments on the PR for further explanation of why I am not able to do that.

This implementation relies on the `Net::Impl` itself to pass the data (or data pointers) using its internal data structures.   This `GLayer` operation only wraps the `forward` function call.  I hope this answer your question.